server/main.lua

```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
+        return
+    end
+
+	local jobName = player.PlayerData[data.group].name
+	local logArea = data.group == 'gang' and 'gang' or 'boss'
+
+    local success = data.group == 'gang' and target.Functions.SetGang(jobName, data.grade) or target.Functions.SetJob(jobName, data.grade)
+    local grade = data.group == 'gang' and gangs[jobName].grades[data.grade].name or jobs[jobName].grades[data.grade].name
+
+    if success then
+        local playerFullName = player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname
+        local targetFullName = target.PlayerData.charinfo.firstname .. ' ' .. target.PlayerData.charinfo.lastname
+        local organizationLabel = player.PlayerData[data.group].label
+
+        exports.qbx_core:Notify(source, 'You hired ' .. targetFullName .. ' into ' .. organizationLabel, 'success')
+        exports.qbx_core:Notify(target.PlayerData.source, 'You have been hired into ' .. organizationLabel, 'success')
+        TriggerEvent('qb-log:server:CreateLog', logArea..'menu', grade, 'yellow', playerFullName.. ' successfully recruited ' .. targetFullName .. ' (' .. organizationLabel .. ')', false)
+    else
+        exports.qbx_core:Notify(source, 'Couldn\'t hire civilian', 'error')
+    end
+	TriggerClientEvent('qbx_management:client:openMenu', source, data.group)
+end)
+
+---@param data {source: number, group: 'job'|'gang'}
+RegisterNetEvent('qbx_management:server:fireEmployee', function(data)
+	local src = source
+	local player = exports.qbx_core:GetPlayer(src)
+	local employee = exports.qbx_core:GetPlayerByCitizenId(data.source) or nil
+	local playerFullName = player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname
+	local organizationLabel = player.PlayerData[data.group].label
+	local logArea = data.group == 'gang' and 'gang' or 'boss'
+	
+	if not player.PlayerData[data.group].isboss then exploitBan(src, 'FireEmployee Exploiting') return end
+	
+	local logType = data.group == 'gang' and 'Gang Member fired!' or 'Employee fired!'
+	if employee then
+		local employeeFullName = employee.PlayerData.charinfo.firstname .. ' ' .. employee.PlayerData.charinfo.lastname
+		if employee.PlayerData.citizenid == player.PlayerData.citizenid then
+			local message = data.group == 'gang' and 'You can\'t kick yourself out of the gang!' or 'You can\'t fire yourself'
+			exports.qbx_core:Notify(src, message, 'error')
+			return
+		end
+	
+		if employee.PlayerData[data.group].grade.level > player.PlayerData[data.group].grade.level then
+			exports.qbx_core:Notify(src, 'You cannot fire your boss!', 'error')
+			return
+		end
+	
+		local success = data.group == 'gang' and employee.Functions.SetGang('none', 0) or employee.Functions.SetJob('unemployed', 0)
+		if not success then
+			exports.qbx_core:Notify(src, 'Unable to fire citizen.', 'error')
+			return
+		end
+	
+		local notifyMessage = data.group == 'gang' and 'You have been expelled from the gang!' or 'You have been fired! Good luck.'
```

Comment:
Was this previously not localized?

Reply:
No. There was no locales in this resource.

Reply:
No there wasn't any translations in this resource.